### PR TITLE
feat: use event name table map

### PR DESCRIPTION
**Pull request by lvrach**
### Motivation
A prospect customer stumble upon an incompatibility between segment and us, on mapping event names to table names.

We map `omega v2` as `omega_v_2`, while segment does `omega_v2`. Although, the second mapping makes more sense, we can not change the behaviour in our system for backwards compatibility reasons. 

We want to introduce a new mechanism, to workaround this problem for the customer, but also address similar issues when a custom mapping is need.

Ideally, customers should be able to do this for control-plane during destination configuration. 

### Solution

This PR provides a way to quickly support custom table name mappings, using an environment variables.

`WAREHOUSE_EVENT_NAME_TABLE_MAP={"<destinationID>": {"omega v2": "omega_v2"}}`

Until, we develop a way to support this from control-plane, destination settings.

### Considerations

Existing ways for passing configuration:

1. Pass them as integration options on user transformations, like we have done for disabling user and track tables. -> It will pollute user transformation, since we are not sure how we want to handle product wise. 
2. Pass it as query param from rudder-server. Will require changes in two systems, passing a mapping as query param could be more complex and error prone. We could consider passing it in the JSON payload, but it would pollute rudder-server even further.

## What is the related Linear task?

Resolves INT-XXX

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3582/test-report.html

File: src/warehouse/v1/util.js [hunk 1]
```diff
@@ -4,10 +4,18 @@ const reservedANSIKeywordsMap = require('../config/ReservedKeywords.json');
 const { isDataLakeProvider } = require('../config/helpers');
 const { TransformationError } = require('@rudderstack/integrations-lib');
 
+let eventNameTableMap = JSON.parse(process.env.WAREHOUSE_EVENT_NAME_TABLE_MAP ?? '{}');
+
 function safeTableName(options, name = '') {
   const { provider } = options;
```
Comment (koladilip) [inline src/warehouse/v1/util.js:10]: `destinationId` can be defined here

**Review by koladilip — DISMISSED** [review]

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Comment by devops-github-rudderstack** [comment]
This PR is considered to be stale. It has been open for 20 days with no further activity thus it is going to be closed in 7 days. To avoid such a case please consider removing the stale label manually or add a comment to the PR.

**Pull request closed without merging**